File: sort/0349_intersection_of_two_arrays/0349_intersection_of_two_arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        if len(nums1) > len(nums2):
            return self.intersection(nums2, nums1)

        lookup = set()
        for i in nums1:
            lookup.add(i)

        res = []
        for i in nums2:
            if i in lookup:
                res += i,
                lookup.discard(i)
        
        return res

nums1 = [1,2,2,1]
nums2 = [2,2]
res = Solution().intersection(nums1, nums2)
print(res)

File: sort/0349_intersection_of_two_arrays/0349_intersection_of_two_arrays.go
package main 

import "fmt"

func intersection(nums1 []int, nums2 []int) []int {
	m := map[int]bool{}
	var res []int
	for _, n := range nums1{
		m[n] = true
	}
	for _, n := range nums2{
		if m[n] {
			delete(m,n)
			res = append(res, n)
		}
	}

	return res
}

func main() {
	nums1 := []int{1,2,2,1}
	nums2 := []int{2,2}
	res := intersection(nums1, nums2)
	fmt.Println(res)
}